outage/2024-10-09-ota-lite.md

```diff
@@ -0,0 +1,57 @@
+## OTA Lite Outage
+
+OTA-lite was unavailble from 20:13UTC to 20:35UTC. During this time things
+like publishing TUF Targets and managing devices (from user facing APIs)
+was not possible. The device-gateway remained up during this outage.
+
+### What Happened
+
+On the previous day, 2024-10-08, we performed a task where we periodically
+rotate secure keys and tokens used by our backend. After rotating keys, we
+restarted each container affected by the key rotation to make sure they
+continuted to function properly. We also deleted some old service accounts
+and tokens no longer used by our system.
+
+We went to bed and slept comfortably.
+
+Later the next day OTA lite would not start up correctly after we tried
+to deploy a very minor change to it. We tried to rollback the change only
+to discover it couldn't run either.
+
+The failure logs pointed at the fact we were using a key that didn't exist.
+
+The missing key was for a minor part of the system where we keep backups of
+each version of Targets.json a customer publishes. It's not stricly required,
+so the quick fix was to comment out this code to give us breathing room.
+
+The root cause wound up being a combination of a bad/naive Grep expression
+combined with a bad variable naming. We were referencing a Kubernetes secret
+in a Pod with:
+```
+      - name: gce-creds
+        secret:
+          defaultMode: 420
+          secretName: ota-api-artifacts
+```
+
+Our grepping looked for `gce-creds` and led us to believe we needed to rotated
+the `gce-creds` key. Another Pod also happened to use this. We rotated the
+key and restarted both containers and validated things were working.
+Later, we determied the service account token the actual seecret, `ota-api-artifacts`,
+was pointing to was no longer necessary. The Pod continued to work and more
+or less fail silently. It wasn't till tried to roll out a new deployment
+did our start-up code fail due to the invalid key.
```

Comment:
IMHO there is too much detail in this paragraph.
Can we truncate by some 80%, e.g. like below:

```suggestion
The root cause wound up being a combination of a bad/naive Grep expression combined with a bad variable naming.
As a result we de facto rotated not only a secret content, but also a secret name; while the API K8s specs were still pointing to the old secret name.
As the API was not restarted during this change, it continued to work as if nothing changed.
But, at a time of the roll out of a new API version it started failing due to the missing required secret.
```